sound/soc/sof/sof-priv.h

```diff
@@ -431,6 +434,7 @@ struct snd_sof_dev {
 	struct list_head route_list;
 	struct snd_soc_component *component;
 	u32 enabled_cores_mask; /* keep track of enabled cores */
+	int core_refs[SND_SOF_CORE_MAX];
```

Comment:
> have you considered using `struct kref`? Don't you think we might in the future or maybe even already now use some extended per-core status? So we could define a `struct sof_dsp_core` right away even if it would only contain one member for now.

Yes, actually you had suggested this here 2 years back :-) : https://github.com/thesofproject/linux/pull/888#pullrequestreview-241579051

Let me change to use kref.